fiftyone/utils/super_gradients.py

```diff
@@ -0,0 +1,193 @@
+"""
+YOLO-NAS model wrapper for the FiftyOne Model Zoo.
+
+| Copyright 2017-2023, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import logging
+import fiftyone as fo
+import fiftyone.core.utils as fou
+import fiftyone.utils.torch as fout
+import fiftyone.zoo.models as fozm
+
+fou.ensure_torch()
+import torch
+import numpy as np
+
+super_gradients = fou.lazy_import(
+    "super_gradients", callback=lambda: fou.ensure_package("super-gradients")
+)
+
+logger = logging.getLogger(__name__)
+
+
+def convert_super_gradients_model(model):
+    """Converts the given super_gradients model into a FiftyOne model.
+
+    Args:
+        model: an ``super_gradients.training.models.detection_models.yolo_nas`` model
+
+    Returns:
+         a :class:`fiftyone.core.models.Model`
+
+    Raises:
+        ValueError: if the model could not be converted
+    """
+    if type(model).__module__.startswith(
+        "super_gradients.training.models.detection_models.yolo_nas"
+    ):
+        return _convert_yolo_nas_detection_model(model)
+    else:
+        raise ValueError(
+            "Unsupported model type; cannot convert %s to a FiftyOne model"
+            % model
+        )
+
+
+def _convert_yolo_nas_detection_model(model):
+    """Converts a YOLO-NAS model from the Super-Gradients library to a FiftyOne compatible format using the TorchYoloNasModel.
+
+    Args:
+        model: An instance of a YOLO-NAS model from the Super-Gradients library.
+
+    Returns:
+        An instance of TorchYoloNasModel configured and ready for use within FiftyOne workflows, preserving the original YOLO-NAS model's characteristics and settings.
+    """
+    config_model = {
+        "labels_path": "{{eta-resources}}/ms-coco-labels.txt",
+        "output_processor_cls": "fiftyone.utils.torch.ClassifierOutputProcessor",
+        "raw_inputs": True,
+    }
+
+    config = TorchYoloNasModelConfig(config_model)
+    return TorchYoloNasModel(config, preloaded_model=model)
```

Comment:
The pattern that we've used in similar situations here is to add an optional `model` parameter to the `Config` class rather than passing it directly to the `Model` class. By convention, the `Model` constructors are supposed to accept only a single `config` argument.

Here's how that looks for Ultralytics YOLO:
https://github.com/voxel51/fiftyone/blob/f902efb2e6dfc0ceb417c9127edf1551cfc08c1b/fiftyone/utils/ultralytics.py#L294

Could you make that small change?

Reply:
Hi @brimoor 
Yes, I already made the change 💪